hooks/boot/03-foreman-maintain-extensions.rb

```diff
@@ -0,0 +1,43 @@
+require 'English'
+require 'open3'
+
+module ForemanMaintainHookContextExtension
+  def package_lock_feature?
+    foreman_maintain('packages -h')
+  end
+
+  def packages_locked?
+    foreman_maintain('packages is-locked --assumeyes')
+  end
+
+  def lock_packages
+    foreman_maintain('packages lock --assumeyes', true)
+  end
+
+  def unlock_packages
+    foreman_maintain('packages unlock --assumeyes', true)
+  end
+
+  def start_services(services)
+    foreman_maintain("service start --only=#{services.join(',')}", true)
+  end
+
+  def stop_services(services)
+    command = "service stop"
+    command = "service stop --only=#{services.join(',')}" if services
+
+    foreman_maintain(command, true)
+  end
+
+  def foreman_maintain(command, allow_fail = false)
```

Comment:
Allow fail feels like an odd naming. My initial understanding of it was that it allowed to continue but the implementation is quite the opposite. How about `exit_on_fail`?